src/Client.ts

```diff
@@ -74,6 +77,20 @@ export default class Client implements ClientBase {
 		this.socket.on(
 			ClientSocketEvents.JoinRoom,
 			async (payload: { name: string; backendModel: BackendModel }) => {
+				// Prevent users from spamming the join button
+				const current_time = Date.now();
+				if (
+					this.connected_at !== null &&
+					current_time - this.connected_at < JOIN_TIMEOUT
```

Comment:
I think `connected_at` shouldn't be optional in the class & should be instantiated as `0` as you have a bit of unreliable code that will result in the following being ran:
`(Date.now() - undefined) < 5` -> `NaN < 5` which is always `false`.

Reply:
good catch